src/windows.rs

```diff
@@ -0,0 +1,343 @@
+#![allow(unknown_lints, bare_trait_objects)]
+
+//! Signal emulation for Windows.
+
+use libc::c_int;
+use std::collections::{BTreeMap, HashMap};
+use std::io::Error;
+use std::sync::Mutex;
+use winapi::shared::minwindef::{BOOL, DWORD, FALSE, TRUE};
+use winapi::um::consoleapi::SetConsoleCtrlHandler;
+use winapi::um::wincon::{CTRL_BREAK_EVENT, CTRL_C_EVENT};
+
+// From Windows CRT...
+// SIGINT = 2
+// SIGILL = 4
+// SIGFPE = 8
+// SIGSEGV = 11
+// SIGTERM = 15
+// SIGBREAK = 21
+// SIGABRT = 22
+pub use libc::{SIGABRT, SIGFPE, SIGILL, SIGINT, SIGSEGV, SIGTERM};
+
+const DUMMY_OFFSET: c_int = 64;
+
+// Define as Linux x86 signal number + 10000
+/// Dummy value for SIGALRM which is missing in Windows.
+pub const SIGALRM: c_int = DUMMY_OFFSET + 14;
+/// Dummy value for SIGBUS which is missing in Windows.
+pub const SIGBUS: c_int = DUMMY_OFFSET + 7;
+/// Dummy value for SIGCHLD which is missing in Windows.
+pub const SIGCHLD: c_int = DUMMY_OFFSET + 17;
+/// Dummy value for SIGCONT which is missing in Windows.
+pub const SIGCONT: c_int = DUMMY_OFFSET + 18;
+/// Dummy value for SIGHUP which is missing in Windows.
+pub const SIGHUP: c_int = DUMMY_OFFSET + 1;
+/// Dummy value for SIGIO which is missing in Windows.
+pub const SIGIO: c_int = DUMMY_OFFSET + 29;
+/// Dummy value for SIGKILL which is missing in Windows.
+pub const SIGKILL: c_int = DUMMY_OFFSET + 9;
+/// Dummy value for SIGPIPE which is missing in Windows.
+pub const SIGPIPE: c_int = DUMMY_OFFSET + 13;
+/// Dummy value for SIGPROF which is missing in Windows.
+pub const SIGPROF: c_int = DUMMY_OFFSET + 27;
+/// Dummy value for SIGQUIT which is missing in Windows.
+pub const SIGQUIT: c_int = DUMMY_OFFSET + 3;
+/// Dummy value for SIGSTOP which is missing in Windows.
+pub const SIGSTOP: c_int = DUMMY_OFFSET + 19;
+/// Dummy value for SIGSYS which is missing in Windows.
+pub const SIGSYS: c_int = DUMMY_OFFSET + 31;
+/// Dummy value for SIGTRAP which is missing in Windows.
+pub const SIGTRAP: c_int = DUMMY_OFFSET + 5;
+/// Dummy value for SIGUSR1 which is missing in Windows.
+pub const SIGUSR1: c_int = DUMMY_OFFSET + 10;
+/// Dummy value for SIGUSR2 which is missing in Windows.
+pub const SIGUSR2: c_int = DUMMY_OFFSET + 12;
+/// Dummy value for SIGWINCH which is missing in Windows.
+pub const SIGWINCH: c_int = DUMMY_OFFSET + 28;
```

Comment:
Can you explain why these constants are interesting and should exist? If I understand it correctly, they are not available on Windows. Therefore, any use of these will likely lead to an error, or at least nothing will happen.

If it's the case, I'd prefer not to have them at all, so the programmer is aware at compile time that functionality is not present, instead of having the program terminate with an error or not work.

Reply:
I tended to this to have "a unified interface" here to minimize `#[cfg]` in implementation and documentation. I started to think that this isn't a good idea, but still want to be able to use `SIGUSR1` in tests and doc-tests.